databases/databases.go

```diff
@@ -191,6 +194,16 @@ func NewDatabase(dbName string) (Database, error) {
 	return db, err
 }
 
+// Returns a list of endpoints that can be used as sources or destinations
+// for transfer with the given database.
+func DatabaseEndpointNames(dbName string) ([]string, error) {
```

Comment:
I might be adopting a more "orthodox" approach to Go here, but I feel like this function could be defined in the context in which it's used, and that might prevent us from noising up this package. Just a thought.